syft/execution/role.py

```diff
@@ -159,31 +186,21 @@ def _store_placeholders(self, obj):
         """
         Replace in an object all FrameworkTensors with Placeholder ids
         """
-        if isinstance(obj, (tuple, list)):
-            r = [self._store_placeholders(o) for o in obj]
-            return type(obj)(r)
-        elif isinstance(obj, dict):
-            return {key: self._store_placeholders(value) for key, value in obj.items()}
-        elif isinstance(obj, PlaceHolder):
+
+        def traversal_function(obj):
             if obj.id.value not in self.placeholders:
                 self.placeholders[obj.id.value] = obj
             return obj.id
-        else:
-            return obj
+
+        return Role.nested_object_traversal(obj, traversal_function, PlaceHolder)
```

Comment:
Wait, isn't it weird to have the object traversal in both `_store_placeholders` and the methods that use it?

Reply:
Performance wise, it's almost the same, but it's awkard, didn't see it, i'll make `register_inputs` and `register_outputs` not use _store_placeholders anymore (I think _store_placeholders could be removed with a `traversal_function` thing, but i'll leave it like this for readability if you are ok with this.